### ISSUE TITLE
Invalid left-hand side in assignment with _getOuter()

### ISSUE THREAD

**Issue by gpbl** [babel/babel#861]
hi! i run into this piece of code:

```js
var test = {};
var other = {
  test: function() {
    test = 1;
  }
};
```
which babel transpiles as
```js
var test = {};
var other = {
  test: (function () {
    function _getOuter() {
      return test;
    }
    return function test() {
      _getOuter() = 1;
    };
  })()
};
```
does that `_getOuter()=1` feel right? Webpack says `Module parse failed: Invalid left-hand side in assignment`. Related to #785

**Comment by sebmck** [comment]
Yeah that's incorrect code. Should be a relatively easy fix, assignmens
aren't briny honorary and the function wrap is to ensure proper scope. /cc
@RReverser

On Sunday, 22 February 2015, Giampaolo Bellavite <notifications@github.com>
wrote:

> hi! i run into this piece of code:
>
> var test = {};var other = {
>   test: function() {
>     test = 1;
>   }
> };
>
> which babel transpiles as
>
> var test = {};var other = {
>   test: (function () {
>     function _getOuter() {
>       return test;
>     }
>     return function test() {
>       _getOuter() = 1;
>     };
>   })()
> };
>
> does that _getOuter()=1 feel right? Webpack says Module parse failed:
> Invalid left-hand side in assignment. Related to #785
> <https://github.com/babel/babel/pull/785>
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/babel/babel/issues/861>.
>


-- 
Sebastian McKenzie

**Comment by sebmck** [comment]
Alright, perhaps this is overengineered but when naming functions it can do it in one of three ways:

 - If there's an own local binding with the same name that's a param or an assignment to the inferred function name then it deoptimises into a long form.
 - If there exists only references to the function name then it turns it into the previous behavour of `getOuter()`.
 - If there are no local references and assignments or an own local binding of the same name then it just sets the function id.

**Comment by sebmck** [comment]
Fixed as of 4.4.4, thanks!
